tests/sentry/api/endpoints/test_project_rule_details.py

```diff
@@ -1552,3 +1558,44 @@ def test_simple(self):
         assert not Rule.objects.filter(
             id=self.rule.id, project=self.project, status=ObjectStatus.PENDING_DELETION
         ).exists()
+
+    @with_feature("organizations:workflow-engine-issue-alert-dual-write")
+    def test_dual_delete_workflow_engine(self):
+        rule = self.create_project_rule(
+            self.project,
+            condition_data=[
+                {
+                    "id": "sentry.rules.conditions.first_seen_event.FirstSeenEventCondition",
+                    "name": "A new issue is created",
+                },
+                {
+                    "id": "sentry.rules.filters.latest_release.LatestReleaseFilter",
+                    "name": "The event occurs",
+                },
+            ],
+        )
+        migrate_issue_alert(rule, user_id=self.user.id)
+
+        alert_rule_workflow = AlertRuleWorkflow.objects.get(rule=rule)
+        workflow = alert_rule_workflow.workflow
+        when_dcg = workflow.when_condition_group
+        if_dcg = WorkflowDataConditionGroup.objects.get(workflow=workflow).condition_group
+
+        assert when_dcg is not None
+        assert if_dcg is not None
+
+        conditions = DataCondition.objects.filter(condition_group=when_dcg)
+        assert conditions.count() == 1
+        filters = DataCondition.objects.filter(condition_group=if_dcg)
+        assert filters.count() == 1
```

Comment:
nit: can skip asserting objects exist if we tested it above, so that we're just testing the delete